tests/regression/56-floats/03-casts.c

```diff
@@ -0,0 +1,59 @@
+// PARAM: --enable ana.int.interval --enable ana.float.interval
+#include <assert.h>
+
+#define RANGE(val, min, max) \
+    if (rnd)                 \
+    {                        \
+        val = min;           \
+    }                        \
+    else                     \
+    {                        \
+        val = max;           \
+    }
+
+int main()
+{
+    // Casts from double into different variants of ints
+    assert((int)0.0);      // FAIL!
+    assert((long)0.0);     // FAIL!
+    assert((unsigned)0.0); // FAIL!
+
+    assert((unsigned)1.0); // SUCCESS!
+    assert((long)2.0);     // SUCCESS!
+    assert((int)3.0);      // SUCCESS!
+
+    // Cast from int into double
+    assert((double)0);  // FAIL!
+    assert((double)0l); // FAIL!
+    assert((double)0u); // FAIL!
+
+    assert((double)1u); // SUCCESS!
+    assert((double)2l); // SUCCESS!
+    assert((double)3);  // SUCCESS!
+
+    // cast with ranges
+    int rnd;
+
+    double value;
+    int i;
+    long l;
+    unsigned u;
+
+    RANGE(i, -5, 5);
+    value = (double)i;
+    assert(-5. <= value && value <= 5.); // SUCCESS!
```

Comment:
Hmm, I get `Assertion "tmp" is unknown" for this line, and similar (just with other variable names) for the lines below...

Reply:
You probably forgot to enable the integer interval representation as well. It is disabled by default but required in those cases, to represent a range. Maybe this can be improved at some point, but as long as `IntegerDomain.maximum` (and also `minimum`) returns not precise enough information when using `def_excl`, those casts can only be successfully done with interval domains.

Reply:
Yes, that was it, thanks! I didn't think about it. Then all looks great :)